src/Components/ConnectLog/index.js

```diff
@@ -1,66 +1,171 @@
 import { fetchLog } from '../../store/actions';
 import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
 import { SkeletonTable } from '@redhat-cloud-services/frontend-components/SkeletonTable';
+import { TableToolbar } from '@redhat-cloud-services/frontend-components/TableToolbar';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
+import flatMap from 'lodash/flatMap';
+import LogNestedTable from './LogNestedtable';
+import { downloadFile } from '../../utils/helpers';
+import { configApi } from '../../api';
 
 const columns = [
   {
     title: 'Initiated date/time',
+    cellFormatters: [expandable],
   },
   'Initiator',
+  {
+    title: 'Playbook',
+    transforms: [cellWidth(20)],
+  },
 ];
 
-const rowsMapper = (results) =>
-  results.map(({ account, created_at: createdAt, id }) => ({
-    id,
-    cells: [
-      <Fragment key="date">
-        <DateFormat date={new Date(createdAt)} extraTitle="Created at: " />
-      </Fragment>,
-      account,
-    ],
-  }));
+const rowsMapper = (results, opened) =>
+  flatMap(
+    results.map(({ account, created_at: createdAt, id, state }, key) => [
+      {
+        id,
+        isOpen: opened.includes(id),
+        cells: [
+          <Fragment key="date">
+            <DateFormat date={new Date(createdAt)} extraTitle="Created at: " />
+          </Fragment>,
+          account,
+          <Fragment key={`download file-${id}`}>
+            <Button
+              variant="link"
+              isInline
+              onClick={() => {
+                (async () => {
+                  const data = await configApi.getPlaybookById(id);
+                  downloadFile(data);
+                })();
+              }}
+            >
+              Download
+            </Button>
+          </Fragment>,
+        ],
+      },
+      {
+        parent: key * 2,
+        cells: [
+          <Fragment key="nested-table">
+            <LogNestedTable
+              services={{
+                useOpenSCAP: state.compliance_openscap,
+                enableCloudConnector: state.remediations,
+              }}
+              isInsights={state.insights}
+            />
+          </Fragment>,
+        ],
+      },
+    ])
+  );
 
 const ConnectLog = ({ isOpen = false, onClose }) => {
+  const [opened, setOpened] = useState([]);
   const dispatch = useDispatch();
   const logLoaded = useSelector(
     ({ logReducer }) => logReducer?.loaded || false
   );
   const rows = useSelector(({ logReducer }) => logReducer?.results || []);
+  const pagination = useSelector(
+    ({ logReducer }) => ({
+      itemCount: logReducer?.total,
+      perPage: logReducer?.limit,
+      page:
+        Math.floor((logReducer?.offset || 0) / (logReducer?.limit || 0)) + 1,
+    }),
+    shallowEqual
+  );
   useEffect(() => {
     if (isOpen) {
-      dispatch(fetchLog());
+      dispatch(fetchLog({}));
```

Comment:
I would rather see assigning an empty object default value to the `fetchLog` function argument instead of passing it. It's easy to forget.